life_expectancy/tests/test_cleaning.py

```diff
@@ -1,17 +1,64 @@
-"""Tests for the cleaning module"""
+"""Tests for all modules"""
+
 import pandas as pd
+from ..load_save import load_data, save_data
+from ..cleaning import clean_data
+from ..main import main
+from unittest.mock import Mock, patch
+from . import OUTPUT_DIR, FIXTURES_DIR
+
+def _correct_data_types(df: pd.DataFrame) -> pd.DataFrame:
+    """Set correct data types.
+    
+    Args:
+        df (pd.DataFrame): The data.
+
+    Returns:
+        df (pd.DataFrame): The modified data.
+    """
+    for col in ['unit', 'sex', 'age', 'region']:
+        df[col] = df[col].astype('string')
+    df['year'] = df['year'].astype('int')
+    df['value'] = pd.to_numeric(
+        df['value'])
+    return df
+
+def test_clean_data(pt_life_expectancy_input: pd.DataFrame,
+                     pt_life_expectancy_cleaned_expected: pd.DataFrame):
+    """Test for the clean_data function
+    
+    Args:
+        pt_life_expectancy_input (pd.DataFrame): Fixture for function input.
+        pt_life_expectancy_cleaned_expected (pd.DataFrame): Fixture for function output.
+
+    """
+
+    pt_life_expectancy_cleaned_actual = clean_data(pt_life_expectancy_input, 'PT')
+    pd.testing.assert_frame_equal(pt_life_expectancy_cleaned_actual,
+                                  _correct_data_types(pt_life_expectancy_cleaned_expected))
 
-from life_expectancy.cleaning import main
-from . import OUTPUT_DIR
+@patch(".load_save.save_data.pandas.DataFrame.to_csv")
+def test_save_data(to_csv_mock: Mock, pt_life_expectancy_cleaned_expected: pd.DataFrame):
+    """Test for the save_data function
+    
+    Args:
+        pt_life_expectancy_cleaned_expected (pd.DataFrame): Fixture for function output.
 
+    """
+    def _print_message(*args, **kwargs):
+        print('Saved file')
+    to_csv_mock(side_effect= _print_message())
+    save_data(pt_life_expectancy_cleaned_expected, 'PT')
+    to_csv_mock.assert_called_once_with('pt_life_expectancy.csv', sep=',', index=False)
```

Comment:
```suggestion
    save_data(pt_life_expectancy_cleaned_expected, 'PT')
    
    BASE_DIR = Path().cwd() / 'life_expectancy' / 'data'
    
    expected_path = BASE_DIR / 'pt_life_expectancy.csv'
    to_csv_mock.assert_called_once_with(expected_path, index=False)
```
Your assertion error was coming from the way you set your path on the `save_data()` and I think you ended up comparing PurePath with PosixPath ([more about it on this page](https://www.freecodecamp.org/news/how-to-use-pathlib-module-in-python/))

What I did was to set it the same way as in the source function and it solves the question!

Reply:
Thank you Nuno!